### ISSUE TITLE
Move 'receiver' out of FETools::extrapolate() CellData structure

### ISSUE THREAD

**Issue by bangerth** [dealii/dealii#13243]
Once #13212 is merged, I will do one more change: Currently, the `CellData` structure stores a `receiver` element that is an MPI rank. But this is conceptually not part of the "cell data", and it is also used after receiving data from another process to indicate the *sender*. A better data structure would be a map (or better multimap) mapping from the MPI rank to the `CellData`.

This is simply a github issue to remind myself that that should be done.

**Comment by tamiko** [comment]
@bangerth Any motivation to work on this? :-)
